ChangeFeedLabSolution/ChangeFeedProcessor.cs

```diff
@@ -10,65 +10,110 @@
+        }
 
         /// <summary>
         /// Processes modified records from Cosmos DB Collection into the Event Hub.
         /// </summary>
         /// <param name="documents"> Modified records from Cosmos DB collections. </param>
         /// <param name="log"> Outputs modified records to Event Hub. </param>
         [FunctionName("ChangeFeedProcessor")]
-        public static void Run(
+        public async Task Run(
             //change database name below if different than specified in the lab
             [CosmosDBTrigger(databaseName: "changefeedlabdatabase",
             //change the collection name below if different than specified in the lab
             collectionName: "changefeedlabcollection",
             ConnectionStringSetting = "DBconnection",
             LeaseConnectionStringSetting = "DBconnection",
             LeaseCollectionName = "leases",
-            CreateLeaseCollectionIfNotExists = true)]IReadOnlyList<Document> documents, TraceWriter log)
+            CreateLeaseCollectionIfNotExists = true)]IReadOnlyList<Document> documents, ILogger log)
         {
-            // Create variable to hold connection string to enable event hub namespace access.
-#pragma warning disable CS0618 // Type or member is obsolete
-            string eventHubNamespaceConnection = ConfigurationSettings.AppSettings["EventHubNamespaceConnection"];
-#pragma warning restore CS0618 // Type or member is obsolete
+            var batches = default(IEnumerable<EventDataBatch>);
+            var eventsToSend = new Queue<EventData>();
 
-            // Build connection string to access event hub within event hub namespace.
-            EventHubsConnectionStringBuilder eventHubConnectionStringBuilder =
-                new EventHubsConnectionStringBuilder(eventHubNamespaceConnection)
+            try
+            {
+                foreach (var doc in documents)
+                {
+                    string json = JsonSerializer.Serialize(doc);
+                    EventData data = new EventData(json);
+                    eventsToSend.Enqueue(data);
+                }
+
+                batches = await BuildBatchesAsync(eventsToSend, eventHubProducerClient);
+                foreach (var batch in batches)
+                {
+                    await eventHubProducerClient.SendAsync(batch);
+                    batch.Dispose();
+                }
+            }
+            finally
+            {
+                foreach (EventDataBatch batch in batches ?? Array.Empty<EventDataBatch>())
                 {
-                    EntityPath = EventHubName
-                };
+                    batch.Dispose();
+                }
 
-            // Create event hub client to send change feed events to event hub.
-            EventHubClient eventHubClient = EventHubClient.CreateFromConnectionString(eventHubConnectionStringBuilder.ToString());
+                await eventHubProducerClient.CloseAsync();

```

Comment:
Have we actually run this code?

The producer is registered with DI as a singleton;  if you close it here, it can no longer be used and subsequent invocations of the function will fail.